compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -440,13 +440,29 @@ impl<'interner> Monomorphizer<'interner> {
             HirExpression::Block(block) => self.block(block.statements)?,
 
             HirExpression::Prefix(prefix) => {
+                let rhs = self.expr(prefix.rhs)?;
                 let location = self.interner.expr_location(&expr);
-                ast::Expression::Unary(ast::Unary {
-                    operator: prefix.operator,
-                    rhs: Box::new(self.expr(prefix.rhs)?),
-                    result_type: Self::convert_type(&self.interner.id_type(expr), location)?,
-                    location,
-                })
+
+                if self.interner.get_selected_impl_for_expression(expr).is_some() {
+                    // If an impl was selected for this prefix operator, replace it
+                    // with a method call to the appropriate trait impl method.
+                    let (function_type, ret) =
+                        self.interner.get_prefix_operator_type(expr, prefix.rhs);
+
+                    let method = prefix
+                        .trait_method_id
+                        .expect("ice: missing trait method if when impl was found");
+                    let func = self.resolve_trait_method_expr(expr, function_type, method)?;
+                    self.create_prefix_operator_impl_call(func, rhs, ret, location)?
+                } else {
+                    let rhs = Box::new(rhs);
+                    ast::Expression::Unary(ast::Unary {
+                        operator: prefix.operator,
+                        rhs,
+                        result_type: Self::convert_type(&self.interner.id_type(expr), location)?,
```

Comment:
If you create let bindings for these two fields, I think the struct should fit on one line